e2e/ibc_to_cctp_test.go

```diff
@@ -21,146 +21,326 @@
 package e2e
 
 import (
+	"context"
 	"strconv"
 	"testing"
 
 	"github.com/strangelove-ventures/interchaintest/v8/ibc"
+	interchainutil "github.com/strangelove-ventures/interchaintest/v8/testutil"
 	"github.com/stretchr/testify/require"
 
 	"cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/codec"
 	transfertypes "github.com/cosmos/ibc-go/v8/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
 
 	"orbiter.dev"
 	"orbiter.dev/testutil"
 	"orbiter.dev/types"
-	"orbiter.dev/types/controller/action"
-	"orbiter.dev/types/controller/forwarding"
+	actiontypes "orbiter.dev/types/controller/action"
+	forwardingtypes "orbiter.dev/types/controller/forwarding"
 	"orbiter.dev/types/core"
 )
 
-const OrbiterModuleAddr = "noble15xt7kx5mles58vkkfxvf0lq78sw04jajvfgd4d"
-
-func TestIbc(t *testing.T) {
+func TestIBCToCCTP(t *testing.T) {
 	t.Parallel()
 
-	// ARRANGE
+	testutil.SetSDKConfig()
 	ctx, s := NewSuite(t, true, true)
 
-	orbiterToCounterpartyChannelID, counterpartyToOrbiterChannelID := s.GetChannels(t, ctx)
+	fromOrbiterChannelID, toOrbiterChannelID := s.GetChannels(t, ctx)
+
+	srcUsdcTrace := transfertypes.ParseDenomTrace(
+		transfertypes.GetPrefixedDenom("transfer", toOrbiterChannelID, Usdc),
+	)
+	dstUsdcDenom := srcUsdcTrace.IBCDenom()
+
+	// Register the interfaces to enable marhsal/unmarshal
+	encCfg := testutil.MakeTestEncodingConfig("noble")
+	orbiter.RegisterInterfaces(encCfg.InterfaceRegistry)
+
+	amountToSend := math.NewInt(OneE6)
+
+	ibcRecipient := s.IBC.CounterpartySender.FormattedAddress()
+	s.FundIBCRecipient(t, ctx, amountToSend, ibcRecipient, fromOrbiterChannelID, dstUsdcDenom)
+
+	t.Run("FailingWithoutForwarding", func(t *testing.T) {
+		testIbcFailingWithoutForwarding(
+			t,
+			ctx,
+			encCfg.Codec,
+			&s,
+			dstUsdcDenom,
+			toOrbiterChannelID,
+		)
+	})
+
+	t.Run("FailingUnsupportedAction", func(t *testing.T) {
+		testIbcFailingUnsupportedAction(
+			t,
+			ctx,
+			encCfg.Codec,
+			&s,
+			dstUsdcDenom,
+			toOrbiterChannelID,
+		)
+	})
+
+	t.Run("PassingWithFeeAction", func(t *testing.T) {
+		testIbcPassingWithFeeAction(
+			t,
+			ctx,
+			encCfg.Codec,
+			&s,
+			dstUsdcDenom,
+			toOrbiterChannelID,
+		)
+	})
+
+	s.FundIBCRecipient(t, ctx, amountToSend, ibcRecipient, fromOrbiterChannelID, dstUsdcDenom)
+	s.FundRecipient(t, ctx, amountToSend, OrbiterModuleAddr)
 
+	t.Run("PassingWithoutActions", func(t *testing.T) {
+		testIbcPassingWithoutActions(
```

Comment:
why do these accounts need to be funded here? could be good to add a comment explaining that